rotkehlchen/db/updates.py

```diff
@@ -0,0 +1,113 @@
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+LAST_DATA_UPDATES_KEY = 'last_data_updates_ts'
+SPAM_ASSETS_URL = 'https://raw.githubusercontent.com/rotki/data/{branch}/updates/spam_assets/assets_v{version}.json'  # noqa: E501
+
+
+class UpdateType(SerializableEnumValueMixin):
+    SPAM_ASSETS = 'spam_assets'
+    RPC_NODES = 'rpc_nodes'
+    CONTRACTS = 'contracts'
+
+
+class RotkiDataUpdater:
+    """
+    Handle updates from the rotki repository related to data that needs to be provided to
+    the users. It includes:
+    - Spam assets
+    - RPC nodes
+    - Contracts
+    """
+
+    def __init__(self, msg_aggregator: 'MessagesAggregator', user_db: 'DBHandler') -> None:
+        self.msg_aggregator = msg_aggregator
+        self.user_db = user_db
+        self.branch = 'main'
+        if not getattr(sys, 'frozen', False):
+            # not packaged -- must be in develop mode
+            self.branch = 'develop'
+
+    def _get_remote_info_json(self) -> dict[str, Any]:
+        """Retrieve remote file with information for different updates"""
+        url = f'https://raw.githubusercontent.com/rotki/data/{self.branch}/updates/info.json'
+        try:
+            response = requests.get(url=url, timeout=DEFAULT_TIMEOUT_TUPLE)
+        except requests.exceptions.RequestException as e:
+            raise RemoteError(f'Failed to query {url} during assets update: {str(e)}') from e
+
+        try:
+            json_data = response.json()
+        except json.decoder.JSONDecodeError as e:
+            raise RemoteError(
+                f'Could not parse update info from {url} as json: {response.text}',
+            ) from e
+
+        return json_data
+
+    def update_spam_assets(self, local_version: int, to_version: int) -> None:
+        """
+        Check the updates in the inclusive range [local_version + 1, to_version] and update the
+        spam assets for those versions. Assets are also added to the globaldb if they don't exist
+        locally
+        """
+        for version in range(local_version + 1, to_version + 1):
+            file_url = SPAM_ASSETS_URL.format(branch=self.branch, version=version)
+            try:
+                updates = query_file(file_url, is_json=True)
+            except RemoteError as e:
+                log.warning(f'Failed to update spam assets due to {str(e)}')
+                break
+            try:
+                if len(updates['assets']) != 0:
+                    log.info(f'Applying update for spam assets from {version}. {len(updates)} tokens to add')  # noqa: E501
+                    with GlobalDBHandler().conn.critical_section():
+                        # Use a critical section to avoid another greenlet adding spam assets at
+                        # the same time
+                        update_spam_assets(db=self.user_db, assets_info=updates['assets'])
+            except KeyError as e:
```

Comment:
nitpick: When I ask to handle the key errors, if it's just one place, keep it only there and if it's just one key you can also just do:

```python
updated_assets = updates.get('assets')
if updated_assets is None:
    log.error(f'Remote update {file_url} does not contain the assets key')
    continue
```

Keep the areas under `try` as small as possible. It's cleaner and shows the potential errors surface clearly